sdk/java-v2/src/main/java/org/arvados/client/logic/collection/FileToken.java

```diff
@@ -42,7 +42,7 @@ public String getFullPath() {
         return Strings.isNullOrEmpty(path) ? fileName : path + fileName;
     }
 
-    public int getFilePosition() {
+    public long getFilePosition() {
```

Comment:
This is a breaking change to the public API, so we may need to introduce a new method instead.

Having said that, any code using the existing method and storing the result in an int is arguably broken and needs to be updated, so perhaps we'd be doing people a favor by making that obvious.

If we go that route, we'll need an update to the readme/release notes with a description of the incompatible API change.

Reply:
The one situation I could see this causing a problem would be if someone was assigning the result of this to an `int`, which would begin to produce a warning or error if the return type is changed to `long`.  However, the original code is clearly wrong, so I don't think backwards compatibility over such a small change is a sufficient argument to preserve an incorrect method signature.